### DISCUSSION TITLE
Enum.ToString() performance

### DISCUSSION THREAD

**Discussion by giladfrid009** [dotnet/runtime#57876] [Ideas] [answered]
I've noticed that ``Enum.ToString()`` method performs a binary search among the enum values in order to print the correct one - resulting in both memory allocations and time complexity of O(log(n)).
Is there a reason to not generate (with source generators) a simple switch clause on the enum values with cases for each possible value, since they are all known compile time - and return ``nameof(case)``?
The switch statement will result in no allocations, and will be translated into jump statements with O(1) time complexity.

**Comment by giladfrid009** [comment, marked as answer]
closing discussion since found a relevant issue.

**Reply by Mrxx99** [reply]
Can you link that issue?

**Reply by MarkPflug** [reply]
I'm assuming it is #57748
